src/main/java/org/jfrog/hudson/util/converters/ArtifactoryBuilderConverter.java

```diff
@@ -0,0 +1,63 @@
+package org.jfrog.hudson.util.converters;
+
+import com.google.common.collect.Lists;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import hudson.util.XStream2;
+import org.jfrog.hudson.ArtifactoryBuilder;
+import org.jfrog.hudson.ArtifactoryServer;
+import org.jfrog.hudson.JFrogPlatformInstance;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class ArtifactoryBuilderConverter extends XStream2.PassthruConverter<ArtifactoryBuilder.DescriptorImpl> {
+    Logger logger = Logger.getLogger(ArtifactoryBuilderConverter.class.getName());
+    List<String> converterErrors = Lists.newArrayList();
+
+    public ArtifactoryBuilderConverter(XStream2 xstream) {
+        super(xstream);
+    }
+
+    @Override
+    protected void callback(ArtifactoryBuilder.DescriptorImpl artifactoryBuilder, UnmarshallingContext unmarshallingContext) {
+        jfrogPlatformMigration(artifactoryBuilder);
+        if (!converterErrors.isEmpty()) {
+            logger.info(converterErrors.toString());
+        }
+    }
+
+    private void jfrogPlatformMigration(ArtifactoryBuilder.DescriptorImpl artifactoryBuilder) {
+        Class<? extends ArtifactoryBuilder.DescriptorImpl> overriderClass = artifactoryBuilder.getClass();
+        Field artifactoryServersField = null;
+        try {
+            artifactoryServersField = overriderClass.getDeclaredField("artifactoryServers");
+            Field jfrogInstancesField = overriderClass.getDeclaredField("jfrogInstances");
+            artifactoryServersField.setAccessible(true);
+            jfrogInstancesField.setAccessible(true);
+
+            Object artifactoryServersObj = artifactoryServersField.get(artifactoryBuilder);
+            Object jfrogServersObj = jfrogInstancesField.get(artifactoryBuilder);
+
+            if (artifactoryServersObj != null && jfrogServersObj == null) {
+                List<ArtifactoryServer> artifactoryServers = (List<ArtifactoryServer>) artifactoryServersObj;
+                // Must not be null.
+                // Once the artifactory builder will get save in the new form(jfrog instances), it is no longer needed to do a conversion.
+                // In order to identify if a conversion has already made for the first time, we validate if 'jfrogInstances' is found.
+                List<JFrogPlatformInstance> jfrogInstances = new ArrayList<>();
+                for (ArtifactoryServer artifactoryServer : artifactoryServers) {
+                    jfrogInstances.add(new JFrogPlatformInstance(artifactoryServer));
+                }
+                jfrogInstancesField.set(artifactoryBuilder, jfrogInstances);
+            }
```

Comment:
By convention, migration logic like this in Jenkins is typically written in an XStream `readResolve()` method as described in [the Jenkins developer documentation](https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility) (where it does not need to use reflection), but this technically works.

Reply:
Thanks for the feedback!.
```
> but this technically works.
```
I may be missing something but if this works, are we in danger that something will break in the future? or this feedback is just a convention use case.
I'm tם try to prioritize this task.

Reply:
> if this works, are we in danger that something will break in the future? or this feedback is just a convention use case.

The latter.

> I'm to try to prioritize this task.

The priority should be to add JCasC tests to this repository using the [JCasC test harness](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/PLUGINS.md#how-to-test). This will ensure that future changes to this repository do not break JCasC support.